code/missionui/fictionviewer.cpp

```diff
@@ -560,55 +526,46 @@ void fiction_viewer_reset()
 	}
 }
 
-void fiction_viewer_load(const char *background_640, const char *background_1024, int ui_index, const char *filename, const char *font_filename, const char *voice_filename)
+void fiction_viewer_load(int stage)
 {
-	int file_length;
-	Assertion(filename, "Invalid fictionviewer filename pointer given!");
-	Assertion(font_filename, "Invalid fictionviewer font filename pointer given!");
-	Assertion(voice_filename, "Invalid fictionviewer voice filename pointer given!");
+	Assertion(stage >= 0 && static_cast<size_t>(stage) < Fiction_viewer_stages.size(), "stage parameter must be in range of Fiction_viewer_stages!");
 
 	// just to be sure
 	if (Fiction_viewer_text != NULL)
 	{
-		Int3();
-		fiction_viewer_reset();
+		Error(LOCATION, "Fiction viewer text should be NULL here!  Trace out and fix!");
```

Comment:
Given that this is checking for bad code instead of an unrecoverable asset failure, this should be an `Assertion()` instead. I'd probably write it like this:

```c++
Assertion(Fiction_viewer_text == nullptr, "Fiction viewer text should be a null pointer, but instead is '%s'. Trace out and fix!\n", Fiction_viewer_text);
```